lib/rstreamor/request.rb

```diff
@@ -1,43 +1,42 @@
 module Rstreamor
   class Request
-    attr_accessor :request, :file
+    attr_reader :request, :file
 
     def initialize(request, file)
-      self.request = request
-      self.file = file
+      @request = request
+      @file = file
     end
 
     def ranges
-      self.request.headers['HTTP_RANGE'].gsub('bytes=', '').split('-') if self.request.headers['HTTP_RANGE']
+      request.headers['HTTP_RANGE'].gsub('bytes=', '').split('-') if request.headers['HTTP_RANGE']
     end
 
     def upper_bound
-      ranges[1] ? ranges[1].to_i : self.file.data.size
+      ranges[1] ? ranges[1].to_i : file.size
```

Comment:
Here we make sure of using the size we are calculating in `File` instead of recalculating.